appcues/src/main/java/com/appcues/analytics/AnalyticsQueueProcessor.kt

```diff
@@ -0,0 +1,132 @@
+package com.appcues.analytics
+
+import androidx.annotation.VisibleForTesting
+import com.appcues.AppcuesCoroutineScope
+import com.appcues.data.AppcuesRepository
+import com.appcues.data.remote.request.ActivityRequest
+import com.appcues.data.remote.request.EventRequest
+import com.appcues.ui.ExperienceRenderer
+import kotlinx.coroutines.launch
+import java.util.Timer
+import java.util.TimerTask
+import kotlin.concurrent.schedule
+
+internal class AnalyticsQueueProcessor(
+    private val appcuesCoroutineScope: AppcuesCoroutineScope,
+    private val repository: AppcuesRepository,
+    private val experienceRenderer: ExperienceRenderer,
+    private val analyticsQueueScheduler: QueueScheduler,
+) {
+
+    interface QueueScheduler {
+
+        fun schedule(block: () -> Unit)
+        fun cancel()
+    }
+
+    private val pendingActivity = mutableListOf<ActivityRequest>()
+
+    @VisibleForTesting
+    fun queueForTesting(activity: ActivityRequest) {
```

Comment:
this is also helpful for testing, so we can add an activity to the list before we call any method.